src/TuDa.CIMS.Api/Controllers/PurchaseController.cs

```diff
@@ -0,0 +1,98 @@
+﻿﻿using Microsoft.AspNetCore.Mvc;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+
+namespace TuDa.CIMS.Api.Controllers;
+
+[ApiController]
+[Route("api/purchase")]
+public class PurchaseController : ControllerBase
+{
+    private readonly IPurchaseService _purchaseService;
+
+    public PurchaseController(IPurchaseService purchaseService)
+    {
+        _purchaseService = purchaseService;
+    }
+
+    /// <summary>
+    /// Retrieves all purchases from the service and returns them in an appropriate HTTP response.
+    /// </summary>
+    /// <param name="workingGroupId">the unique id of a workinggroup </param>
+    /// <returns>a 200 OK response if the operation is successfully and a 400 BadRequest response if any error occurs </returns>
+    [HttpGet($"{{{nameof(workingGroupId)}:guid}}")]
+    public async Task<IActionResult> GetAllAsync(Guid workingGroupId)
+    {
+        return (await _purchaseService.GetAllAsync(workingGroupId)).Match<IActionResult>(
+            purchases => Ok(purchases),
+            error => BadRequest(error)
+        );
+    }
+
+    /// <summary>
+    /// Retrieves a purchase with the specific id from the service and returns it in an appropriate HTTP response.
+    /// </summary>
+    /// <param name="id">the unique id of the purchase</param>
+    /// <param name="workingGroupId">the unique id of a workinggroup </param>
+    /// <returns>a 200 OK response if the operation is successfully and a 400 BadRequest response if any error occurs  </returns>
+    [HttpGet($"{{{nameof(id)}:guid}}/{{{nameof(workingGroupId)}:guid}}")]
```

Comment:
🔧 These two should be switched around, because you're choosing the `WorkingGroup` first and after that the `Purchase`